### PR TITLE
Forbid create voucher from diff type of supply

### PR DESCRIPTION
**Pull request by HristiyanG**

### CODE REVIEW

**Review by thcrnk — APPROVED** [review]

**Review by hswopeams — APPROVED** [review]
Changes made to error codes will break the front-end after the contracts are redeployed. Discussed with Nassar and Hris on 2 april 2021. Since front-end doesn't use mocks, it will need the changed contract to test against. Approving this PR with the caveat that the front-end needs to be changed ASAP to match the error codes.

File: test/2_test_fullpath_with_permit.js [hunk 1]
```diff
@@ -1793,6 +1812,25 @@ contract('Cashier and VoucherKernel', async (addresses) => {
           );
         });
 
+        it('[NEGATIVE] Should not create order from a wrong payment type', async () => {
```
Comment (ali2251) [inline test/2_test_fullpath_with_permit.js:1815]: Whats the difference between these tests? they all look exactly the same to me unless i am missing something?

File: contracts/BosonRouter.sol [hunk 2]
```diff
@@ -50,21 +50,21 @@ contract BosonRouter is
      * @notice Acts as a modifier, but it's cheaper. Checking if a non-zero address is provided, otherwise reverts.
      */
     function notZeroAddress(address tokenAddress) private pure {
-        require(tokenAddress != address(0), "0_A"); //zero address
+        require(tokenAddress != address(0), "0A"); //zero address
```
Comment (ali2251) [inline contracts/BosonRouter.sol:53]: Are these changes needed for this PR?

File: contracts/VoucherKernel.sol [hunk 3]
```diff
@@ -310,15 +310,20 @@ contract VoucherKernel is IVoucherKernel, Ownable, Pausable, UsingHelpers {
...
      * @param _issuer          Address of the token's issuer
      * @param _holder          Address of the recipient of the voucher (ERC-721)
+     * @param _paymentMethod   method being used for that particular order that needs to be fulfilled
      * @param _correlationId           ID of the current interaction with the smart contract for a specific user
      */
     function fillOrder(
         uint256 _tokenIdSupply,
         address _issuer,
         address _holder,
+        uint8 _paymentMethod,
```
Comment (ali2251) [inline contracts/VoucherKernel.sol:320]: where is this payment method used?

File: contracts/BosonRouter.sol (hunk 2 above)
Comment (HristiyanG) [reply contracts/BosonRouter.sol:53]: Yes. I think this was discussed last week (Thursday) on the call we had.

File: contracts/VoucherKernel.sol (hunk 3 above)
Comment (HristiyanG) [reply contracts/VoucherKernel.sol:320]: The payment method is a substantial part throughout the whole business logic.

File: test/2_test_fullpath_with_permit.js (hunk 1 above)
Comment (HristiyanG) [reply test/2_test_fullpath_with_permit.js:1815]: Maybe you should have a look at the Builder
